File: Game-1-modular/entities/components/activity_tracker.py
"""Activity tracking component"""


class ActivityTracker:
    def __init__(self):
        self.activity_counts = {
            'mining': 0, 'forestry': 0, 'smithing': 0, 'refining': 0, 'alchemy': 0,
            'engineering': 0, 'enchanting': 0, 'combat': 0
        }

    def record_activity(self, activity_type: str, amount: int = 1):
        if activity_type in self.activity_counts:
            self.activity_counts[activity_type] += amount

    def get_count(self, activity_type: str) -> int:
        return self.activity_counts.get(activity_type, 0)

File: Game-1-Godot/src/Game1.Core/Progression/ActivityTracker.cs
namespace Game1.Core.Progression;

/// <summary>
/// Port of entities/components/activity_tracker.py — fixed-key counters.
/// BUG-COMPATIBLE: record_activity only increments KNOWN keys, so
/// 'fishing' (absent from the dict) is silently dropped, exactly like
/// Python. get_count returns 0 for unknown keys.
/// </summary>
public sealed class ActivityTracker
{
    public readonly Dictionary<string, int> ActivityCounts = new()
    {
        ["mining"] = 0, ["forestry"] = 0, ["smithing"] = 0, ["refining"] = 0,
        ["alchemy"] = 0, ["engineering"] = 0, ["enchanting"] = 0, ["combat"] = 0,
    };

    public void RecordActivity(string activityType, int amount = 1)
    {
        if (ActivityCounts.ContainsKey(activityType))
            ActivityCounts[activityType] += amount;
    }

    public int GetCount(string activityType) =>
        ActivityCounts.GetValueOrDefault(activityType, 0);
}
